include/deal.II/lac/trilinos_tpetra_vector.templates.h

```diff
@@ -643,8 +750,45 @@ namespace LinearAlgebra
 
     template <typename Number>
     void
-    Vector<Number>::compress(const VectorOperation::values /*operation*/)
-    {}
+    Vector<Number>::compress(const VectorOperation::values operation,
+                             const IndexSet               &locally_owned)
+    {
+      if (!compressed)
+        {
+          Tpetra::CombineMode tpetra_operation = Tpetra::ZERO;
+          if (operation == VectorOperation::insert)
+            tpetra_operation = Tpetra::INSERT;
+          else if (operation == VectorOperation::add)
+            tpetra_operation = Tpetra::ADD;
+          else
+            Assert(false, ExcNotImplemented());
+
+          if (locally_owned != IndexSet())
+            {
+              locally_owned_map.reset();
+              locally_owned_map =
+                locally_owned.make_tpetra_map_rcp(mpi_comm(), false);
```

Comment:
Wouldn't the assignment in the second statement imply the reset operation in the first line?

Reply:
Yes, it does.
I removed that unnecessary reset.